src/library_sdl.js

```diff
@@ -1252,7 +1252,9 @@ var LibrarySDL = {
         if (state === null) return;
         // Check only if the timestamp has differed.
         // NOTE: Timestamp is not available in Firefox.
-        if (typeof state.timestamp !== 'number' || state.timestamp !== prevState.timestamp) {
+        // NOTE: Timestamp is currently not properly set for the GearVR controller
+        // on Samsung Internet: it is always zero.
+        if (typeof state.timestamp !== 'number' || state.timestamp !== prevState.timestamp || state.timestamp == 0) {
```

Comment:
how about `|| !state.timestamp`? that would catch the 0 case, but also other weird invalid falsey values, in a more direct way I think

Reply:
Good idea!